src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -464,94 +462,41 @@ async def generate_chroot(request: SetupPyChrootRequest) -> SetupPyChroot:
 
 @rule
 async def get_sources(request: SetupPySourcesRequest) -> SetupPySources:
-    targets = request.targets
-    stripped_srcs_list = await MultiGet(
-        Get(
-            SourceRootStrippedSources,
-            StripSourcesFieldRequest(
-                target.get(Sources),
-                for_sources_types=(PythonSources, ResourcesSources),
-                enable_codegen=True,
-            ),
-        )
-        for target in targets
+    python_sources = await Get(
+        StrippedPythonSources,
+        StrippedPythonSourcesRequest(
+            targets=request.targets, include_resources=False, include_files=False
+        ),
     )
-
-    # Create a chroot with all the sources, and any ancestor __init__.py files that might be needed
-    # for imports to work.  Note that if a repo has multiple exported targets under a single ancestor
-    # package, then that package must be a namespace package, which in Python 3 means it must not
-    # have an __init__.py. We don't validate this here, because it would require inspecting *all*
-    # targets, whether or not they are in the target set for this run - basically the entire repo.
-    # So it's the repo owners' responsibility to ensure __init__.py hygiene.
-    stripped_srcs_digests = [
-        stripped_sources.snapshot.digest for stripped_sources in stripped_srcs_list
-    ]
-    ancestor_init_pys = await Get(AncestorInitPyFiles, Targets, targets)
-    sources_digest = await Get(
-        Digest, MergeDigests((*stripped_srcs_digests, *ancestor_init_pys.digests))
+    all_sources = await Get(
+        StrippedPythonSources,
+        StrippedPythonSourcesRequest(
+            targets=request.targets, include_resources=True, include_files=True
+        ),
     )
```

Comment:
It'd be much better to do this:

```
Get(SourceFiles, AllSourceFilesRequest(request.targets, for_sources_types=[ResourcesSources], enable_codegen=True, strip_source_roots=True)
```

Should this be including `FilesSources`? NB that those won't be stripped.

Reply:
Why would that be better? Isn't this what `StrippedPythonSourcesRequest` is for? Note that we still need `all_sources` below, so we'd have to do a merge of the python sources and resources anyway.

I think this should include FileSources - even though they won't be stripped (and shouldn't be), if the user has some file they want to jam in the dist (e.g., a README) we should let them.

Reply:
Ah I see. Yeah, if you need `AllSources`, then sticking with this is fine.

Reply:
Sounds good on FilesSources. I agree. Consider adding a test case to ensure that we don't regress on handling files.